### ISSUE TITLE
Tests fails on local machine but not in CI

### ISSUE THREAD

**Issue by nojhan** [nojhan/liquidprompt#773]
Some tests fail on my Ubuntu focal, but actually pass on Github's CI.

Shell: Bash 5.0.17(1)-release & Zsh 5.8
Operating system: Linux 5.4.0-139-generic #156-Ubuntu SMP Fri Jan 20 17:27:18 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux, Ubuntu 20.04.5 LTS focal.
Liquid Prompt version: 6480d86

### Steps to Reproduce

1. Run tests.sh on current master 6480d86,
2. Compare with, e.g. PR #772 (forked from master's 6480d86), where all tests pass.

### Failed tests

[tests_773.log](https://github.com/nojhan/liquidprompt/files/10883171/tests_773.log)

```sh
$ cat tests_773.log | grep -B 1 ASSERT
test_bash_preexec_with_LP_ENABLE_TITLE
ASSERT:
test_no_bash_preexec_with_LP_ENABLE_TITLE
ASSERT:
--
test_path_links
ASSERT:su: path element linked to FILE expected:<\[\]home\[\]> but was:<\[\]home\[\]>
ASSERT:telnet: path element not linked expected:<home> but was:<\[\]home\[\]>
--
test_path_links
ASSERT:su: path element linked to FILE expected:<%{%}home%{%}> but was:<%{%}home%{%}>
ASSERT:telnet: path element not linked expected:<home> but was:<%{%}home%{%}>
```

**Comment by nojhan** [comment]
Full log is already in the bug report (the "[tests_773.log](https://github.com/nojhan/liquidprompt/files/10883171/tests_773.log)" link).

Here is the external tool test. [external_tool_tester.log](https://github.com/nojhan/liquidprompt/files/10885268/external_tool_tester.log).

**Comment by nojhan** [comment]
After more extensive tests back home, I see that those tests fails only from within a tmux session.

**Comment by Rycieos** [comment]
> I see that those tests fails only from within a tmux (3.0a-2ubuntu0.4) session

Now that is interesting. That might be a good thing...
I'll have to dig in.

**Comment by Rycieos** [comment]
I can confirm that the bash_preexec test fails only on Tmux. But the path tests are actually failing for me even without Tmux...

**Comment by Rycieos** [comment]
The paths had an issue with exported SSH* environment variables leaking in. Fix is simple.

I also caught an issue with the Unicode characters in the path hyperlink tests. I added a skip condition those those as well.

Your Tmux issue was easy to track down once I knew it was a Tmux thing. Liquidprompt is smart and disables the title feature in screen/tmux if `LP_ENABLE_TITLE_SCREEN` is not set, which it wasn't.

Overall very easy fixes, and all test configuration problems.

I also caught a Zsh issue that was added recently, and fixed that as well in 9f9f177
